src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -96,42 +165,175 @@ std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::frame_velocity(
              << velocity.size()
              << ") does not match the dimension of the translation map (" << Dim
              << ").");
+  // Section for handling overloaded constructor
+  tt::remove_cvref_wrap_t<DataType> radial_function_value(
+      gsl::at(source_coords, 0));
+  if (f_of_r_ == nullptr) {
+    radial_function_value = 1.0;
+  } else {
+    std::array<tt::remove_cvref_wrap_t<DataType>, Dim> distance_to_center{};
+    Scalar<tt::remove_cvref_wrap_t<DataType>> radius(gsl::at(source_coords, 0));
+    get(radius) = 0.0;
+
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(distance_to_center, i) =
+          gsl::at(source_coords, i) - gsl::at(center_, i);
+      radius.get() += square(gsl::at(distance_to_center, i));
+    }
+    radius.get() = sqrt(radius.get());
+    radial_function_value = (*f_of_r_)(radius.get());
+  }
   for (size_t i = 0; i < Dim; i++) {
-    gsl::at(result, i) = make_with_value<tt::remove_cvref_wrap_t<T>>(
-        dereference_wrapper(gsl::at(source_coords, i)), velocity[i]);
+    gsl::at(result, i) = velocity[i] * radial_function_value;
   }
   return result;
 }
 
 template <size_t Dim>
-template <typename T>
-tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>
-Translation<Dim>::jacobian(const std::array<T, Dim>& source_coords) const {
-  return identity<Dim>(dereference_wrapper(source_coords[0]));
+template <typename DataType>
+tnsr::Ij<tt::remove_cvref_wrap_t<DataType>, Dim, Frame::NoFrame>
+Translation<Dim>::jacobian(
+    const std::array<DataType, Dim>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const {
+  if (f_of_r_ == nullptr) {
+    return identity<Dim>(dereference_wrapper(source_coords[0]));
+  } else {
+    std::array<tt::remove_cvref_wrap_t<DataType>, Dim> distance_to_center{};
```

Comment:
This can also be simplified a bit. Especially the `Scalar`. I don't think that's necessary